docs/proposals/observability.md

```diff
@@ -0,0 +1,311 @@
+# Enhancement Proposal-1778: Observability Policy
+
+- Issue: https://github.com/nginxinc/nginx-gateway-fabric/issues/1778
+- Status: Implementable
+
+## Summary
+
+This Enhancement Proposal introduces the `ObservabilityPolicy` API, which allows Application Developers to define settings related to tracing, metrics, or logging at the HTTPRoute level.
+
+## Goals
+
+- Define the Observability policy.
+- Define an API for the Observability policy.
+
+## Non-Goals
+
+- Provide implementation details for implementing the Observability policy.
+
+## Introduction
+
+### Observability Policy
+
+The Observability Policy contains settings to configure NGINX to expose information through tracing, metrics, and/or logging. This is a Direct Policy that is attached to an HTTPRoute or HTTPRoute Rule by an Application Developer. It works in conjunction with a [Gateway Settings](gateway-settings.md) configuration that contains higher level settings to enable Observability at this lower level. The [Gateway Settings](gateway-settings.md) configuration is managed by a Cluster Operator.
+
+Since this policy is attached to an HTTPRoute or HTTPRoute rule, the Observability settings should just apply to the relevant `location` contexts of the NGINX config for that route or rule.
```

Comment:
Actually, we can't yet apply to a route rule, because the `name` field doesn't exist on a rule yet. There's no way to specify it. So we can only attach to a full route.